src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -223,10 +228,20 @@ public void leaveToken(DetailAST ast) {
                 scope = scopeStack.pop().scope;
                 break;
             case TokenTypes.SLIST:
+                final PrevScopeUnitializedVariableData prevScopeUnitializedVariableData =
+                    prevScopeUninitializedVariableStack.peek();
                 if (ast.getParent().getType() != TokenTypes.CASE_GROUP
-                    || findLastToken(ast.getParent().getParent(), TokenTypes.CASE_GROUP,
-                        TokenTypes.SLIST) == ast.getParent()) {
+                    || findLastChildWhichContainsSpecifiedToken(ast.getParent().getParent(),
+                            TokenTypes.CASE_GROUP, TokenTypes.SLIST) == ast.getParent()) {
                     scope = scopeStack.pop().scope;
+                    prevScopeUninitializedVariableStack.pop();
+                }
+
```

Comment:
@Bhavik3 I don't think we should have empty space here, as It should be as close to `case` as possible